### PR TITLE
feat: Added developer_code_chat.ipynb notebook

### PR DESCRIPTION
**Pull request by CharulataShelar**
- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
- [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb [hunk 1]
```diff
@@ -0,0 +1,712 @@
+{
```
Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Can we not make these comments? It's easiest if users can execute. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Same here </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Can users run this clone? Can they access that repo from their projects? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Is there a way folks can inspect the code that we just cloned?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Please remove any reference to projects. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <p>Can we have a bit more information on what the code does?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (erwinh85) [inline gemini/use-cases/retrieval-augmented-generation/developer_code_chat.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    demo.launch(width="80%", share=True, debug=True)
</pre><p>Please clear all the output before submitting the code. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/486/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

**Comment by CharulataShelar** [comment]
Closing this as we have acknowledge the comments in the upcoming PR.

**Pull request closed without merging**